projects/core/src/lib/components/toolbar/geolocation/geolocation.component.ts

```diff
@@ -0,0 +1,110 @@
+  styleUrls: ['./geolocation.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GeolocationComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+  private featureGeom = new Subject<FeatureModel[]>();
+
+  private activeWatch: number | undefined = undefined;
+
+  public hasGeolocation = navigator.geolocation !== undefined;
+  public isWatching = false;
+  public hasFix = false;
+
+  constructor(
+    private store$: Store,
+    private mapService: MapService,
+    private cdr: ChangeDetectorRef,
+  ) { }
+
+  public ngOnInit(): void {
+    this.mapService.renderFeatures$('geolocation-layer', this.featureGeom.asObservable(), f => {
+      if (f.__fid === 'geolocation-point') {
+        return {
+          styleKey: 'geolocation-point-style',
+          zIndex: 1000,
+          pointType: 'circle',
+          pointSize: 7,
+          pointFillColor: '#6236ff',
+          pointStrokeColor: '#ffffff',
+          pointStrokeWidth: 2,
+        };
+      } else {
+        return {
+          styleKey: 'geolocation-shadow-style',
+          zIndex: 999,
+          fillColor: '#6236ff',
+          fillOpacity: 20,
+          strokeColor: '#6236ff',
+          strokeOpacity: 40,
+        };
+      }
+     }).pipe(takeUntil(this.destroyed)).subscribe();
+  }
+
+  public ngOnDestroy(): void {
+    this.destroyed.next(null);
+    this.destroyed.complete();
+  }
+
+  private positionSuccess(pos: GeolocationPosition) {
+    this.mapService.getProjectionCode$()
+      .pipe(take(1))
+      .subscribe(projectionCode => {
+          const point = WKTHelper.transformWKT(`POINT(${pos.coords.longitude} ${pos.coords.latitude})`, 'EPSG:4326', projectionCode);
+          if (point === undefined) {
+              return;
+          }
+
+          const circle = point.replace('POINT', 'CIRCLE').replace(')', ` ${pos.coords.accuracy})`);
+
+          this.featureGeom.next([
+            { __fid: 'geolocation-point', geometry: point, attributes: {} },
+            { __fid: 'geolocation-shadow', geometry: circle, attributes: {} },
+          ]);
+
+          if (!this.hasFix) {
+            this.hasFix = true;
+            this.mapService.zoomTo(point, projectionCode);
+            this.cdr.detectChanges();
+          }
+      });
+  }
+
+  private cancelGeolocation() {
+      if (this.activeWatch !== undefined) {
+        navigator.geolocation.clearWatch(this.activeWatch);
+        this.activeWatch = undefined;
+      }
+
+      this.featureGeom.next([]);
+      this.isWatching = false;
+      this.hasFix = false;
+      this.cdr.detectChanges();
+  }
+
+  private positionError(_e: GeolocationPositionError) {
+      this.cancelGeolocation();
+  }
+
+  public onClick(): void {
+    if (this.activeWatch === undefined) {
+      this.activeWatch = navigator.geolocation.watchPosition(this.positionSuccess.bind(this), this.positionError.bind(this), { enableHighAccuracy: true });
```

Comment:
According to the story we should zoom to the location once, show the location and them remove the icon after 10 seconds. Maybe its better to add `watch` later once we have clear how this should work in our case

Reply:
This should be fixed; I've just made it cap out at 12 seconds (which includes 2 seconds of margin for zooming to the location). The watch stays, as it means that the phone can still correct its location if it gains more accuracy (though the map won't follow after the initial fix)